packages/curve-ui-kit/src/widgets/Disclaimer/Section.tsx

```diff
@@ -0,0 +1,55 @@
+import { ReactNode, Children } from 'react'
+
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+
+const { Spacing } = SizesAndSpaces
+
+export const Header = ({ children }: { children?: ReactNode }) => (
+  <Stack
+    justifyContent="end"
+    sx={{
+      minHeight: '56px', // TODO: place as a constant somewhere.
+      borderBottom: (t) => `1px solid ${t.design.Layer[1].Outline}`,
+      paddingInline: Spacing.md,
+    }}
+  >
+    <Typography variant="headingSBold">{children}</Typography>
+  </Stack>
+)
+
+export const Title = ({ children }: { children?: ReactNode }) => (
+  <Typography variant="headingXsBold" color="textSecondary">
+    {children}
+  </Typography>
+)
+
+export const Paragraph = ({ children }: { children?: ReactNode }) => (
+  <Typography component="div" variant="bodyMRegular" color="textSecondary">
+    {children}
+  </Typography>
+)
+
+export const Section = ({ children }: { children?: ReactNode }) => {
+  const childArray = Children.toArray(children)
+  const title = childArray.find((child) => typeof child === 'object' && 'type' in child && child.type === Title)
+  const paragraphs = childArray.filter((child) => typeof child === 'object' && 'type' in child && child.type !== Title)
```

Comment:
this children magic isn't ideal. We could use a `title` slot for the title element then just use `children` for the rest

Reply:
I couldn't decide. I don't think it's ideal, but deviating from the rest of the style where I use children using a property again looks weird (I had it as a prop before). I really miss named slots like Vue in React. At least the tabs look aesthetically pleasing and easier to modify when it's all slots instead of properties; looks a bit like shadcn 😅 . Even Claude suggested I'd use this for composability instead of keeping the prop.

Reply:
Ok you though hard about this 👌